tests/conftest.py

```diff
@@ -149,6 +149,35 @@ def table_schema_simple() -> Schema:
     )
 
 
+@pytest.fixture(scope="session")
```

Comment:
nit: wydt about adding an NestedField to [table_schema_simple](https://github.com/apache/iceberg-python/pull/1432/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R141-R149) instead of creating a new Schema altogether?

Reply:
Sure, I can make that change and see what happens. I was hesitant because `table_schema_simple` seems to be used by lots of tests so wasn't sure of the implications.

Reply:
+1 if it becomes too tedious, the current test is fine

Reply:
There were ~25 test failures when changing `table_schema_simple`, with some getting into the pyarrow schema conversions which were a bit more than a copy/paste change.

I can probably figure it out, but might take a bit longer until I find time.